rpc/dial.go

```diff
@@ -93,90 +108,161 @@ func dial(
 	dOpts *dialOptions,
 	tryLocal bool,
 ) (ClientConn, bool, error) {
+	if ctx.Err() != nil {
+		return nil, false, ctx.Err()
+	}
+
 	var isJustDomain bool
-	if strings.HasPrefix(address, "unix://") {
+	switch {
+	case strings.HasPrefix(address, "unix://"):
 		dOpts.mdnsOptions.Disable = true
 		dOpts.webrtcOpts.Disable = true
 		dOpts.insecure = true
 		dOpts.disableDirect = false
-	} else {
-		if strings.ContainsRune(address, ':') {
-			isJustDomain = false
-		} else {
-			isJustDomain = net.ParseIP(address) == nil
-		}
+	case strings.ContainsRune(address, ':'):
+		isJustDomain = false
+	default:
+		isJustDomain = net.ParseIP(address) == nil
 	}
 
+	// RSDK-6151: We make concurrent dial attempts via mDNS and WebRTC, taking the first
```

Comment:
nit: don't need ticket number for if not TODO

Reply:
we've been linking some of our code comments back to JIRA tickets to give more context on things work the way they do, for example: https://github.com/viamrobotics/rdk/pull/3467/files#diff-4e03c7046f4cde2fb3cfabe60f410c7cc0380ec31028edec049f6a37cab65cd8R119

Reply:
On second thought, it's not necessary here